compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -2128,18 +2128,24 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
         // suggest adding an explicit lifetime bound to it.
         let generics = self.tcx.generics_of(generic_param_scope);
         // type_param_span is (span, has_bounds)
+        let mut is_synthetic = false;
+        let mut ast_generics = None;
         let type_param_span = match bound_kind {
             GenericKind::Param(ref param) => {
                 // Account for the case where `param` corresponds to `Self`,
                 // which doesn't have the expected type argument.
                 if !(generics.has_self && param.index == 0) {
                     let type_param = generics.type_param(param, self.tcx);
+                    is_synthetic = matches!(
+                        type_param.kind,
+                        GenericParamDefKind::Type { synthetic: true, .. }
+                    );
```

Comment:
```suggestion
                    is_synthetic = type_param.kind.is_synthetic();
```

Reply:
Thanks for reviewing. Changed accordingly.